docs/fsharp/style-guide/formatting.md

```diff
@@ -1069,24 +1067,61 @@ let printVolumes x =
         (convertVolumeImperialPint x)
 ```
 
-For lambda expressions, you may also want to consider placing the body of a lambda expression on a new line, indented by one scope, if it is long enough:
+For lambda expressions, you may also want to consider placing the body of a lambda expression on a new line, indented by one scope, if it is long enough or followed by other arguments:
 
 ```fsharp
 let printListWithOffset a list1 =
     List.iter
         (fun elem ->
              printfn $"A very long line to format the value: %d{a + elem}")
         list1
+```
+
+If the lambda argument is the last argument in a function application, place all arguments until the arrow on the same line.
+
+```fsharp
+Target.create "Build" (fun ctx ->
+    // code
+    // here
+    ())
 
 let printListWithOffsetPiped a list1 =
     list1
-    |> List.iter
-           (fun elem ->
-                printfn $"A very long line to format the value: %d{a + elem}")
+    |> List.iter (fun elem ->
+        printfn $"A very long line to format the value: %d{a + elem}")
+```
+
+Treat match lambda's in a similar fashion.
+
+```fsharp
+functionName arg1 arg2 arg3 (function
+    | Choice1of2 x
+    | Choice2of2 y)
+```
+
+Unless there are many leading or multiline arguments before the lambda. In that case, indent all arguments with one scope.
+
+```fsharp
+functionName 
+    arg1 
+    arg2 
+    arg3 
+    (fun arg4 ->
+        bodyExpr)
+
+functionName 
+    arg1 
+    arg2 
+    arg3 
+    (function
+     | Choice1of2 x
+     | Choice2of2 y)
 ```
 
 If the body of a lambda expression is multiple lines long, you should consider refactoring it into a locally-scoped function.
 
+// TODO: Is https://github.com/dotnet/docs/issues/22834 still relevant?
```

Comment:
I'm not sure of the question here?

Reply:
Sorry @dsyme, the concrete question would be if this indent from the function name rule still applies when the lambda is the last argument.
Imagine if the indent size is 2 spaces.
```fsharp
list1
|> List.iter (fun elem ->
  // should the body start one indent from the pipe
      // or one indent from the `List.iter`?
  printfn $"A very long line to format the value: %d{elem}")
```

Reply:
Thanks, ready for review now.